### PR TITLE
10.2.0-M1 release notes

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: docs/src/main/paradox/migration-guide/migration-guide-10.2.x.md [hunk 1]
```diff
@@ -4,7 +4,15 @@
...
 
-## Akka HTTP 10.1.11 - > 10.2.0
+## Akka HTTP 10.1.11 -> 10.2.0
+
+### Scalatest dependency upgraded to 3.1.0
+
+The Scalatest dependency for akka-http-testkit was upgraded to version 3.1.0. This version is incompatible with previous
+versions. This is relevant for user code if it uses methods from @scaladoc[ScalatestUtils](akka.http.scaladsl.testkit.ScalatestUtils)
+(which are in scope if your test extends from @scaladoc[ScalaTestRouteTest](akka.http.scaladsl.testkit.ScalaTestRouteTest)).
+In this case, the project itself needs to update to use Scalatest >= 3.1.0.
```
Comment (jrudolph) [inline docs/src/main/paradox/migration-guide/migration-guide-10.2.x.md:14]:
```suggestion
In this case, the project itself needs to be updated to use Scalatest >= 3.1.0.
```

File: docs/src/main/paradox/migration-guide/migration-guide-10.2.x.md [hunk 2]
```diff
@@ -4,7 +4,15 @@
 
 See the general @ref[compatibility guidelines](../compatibility-guidelines.md).
```
Comment (jrudolph) [inline docs/src/main/paradox/migration-guide/migration-guide-10.2.x.md:5]:
```suggestion
See the general @ref[compatibility guidelines](../compatibility-guidelines.md).

Under these guidelines, minor version updates are supposed to be binary compatible and drop-in replacements
for former versions under the condition that user code only uses public, stable, non-deprecated API. Especially
libraries should make sure not to depend on deprecated API to be compatible with both 10.1.x and 10.2.x.

If you find an unexpected incompatibility please let us know, so we can check whether the incompatibility is accidental so we might still be able to fix it.
```

File: docs/src/main/paradox/release-notes/10.2.x.md [hunk 3]
```diff
@@ -1,14 +1,83 @@
...
-(which are in scope if your test extends from @scaladoc[ScalaTestRouteTest](akka.http.scaladsl.testkit.ScalaTestRouteTest)).
-In this case, the project itself needs to update to use Scalatest >= 3.1.0.
+See the @ref[Migration guide](../migration-guide/migration-guide-10.2.x.md).
 
 ### Changes since 10.1.11
+
+This page currently contains the changes between 10.1.11 and the first
+milestone release of 10.2.x, 10.2.0-M1. This list will be amended when
+we release 10.2.0 proper.
+
```
Comment (jrudolph) [inline docs/src/main/paradox/release-notes/10.2.x.md:12]: We should feature a few things here:

 * proper 2.6 support
 * support for message attributes
 * link to implicit changes in routing API

File: docs/src/main/paradox/release-notes/10.2.x.md [hunk 4]
```diff
@@ -1,14 +1,83 @@
...
+* Names for symbolic enhancements [#3082](https://github.com/akka/akka-http/pull/3082)
+* Allow names to match paths [#3089](https://github.com/akka/akka-http/pull/3089)
+
+#### akka-http-marshallers
+
+* Jackson: Better JSON validation error when unmarshalling [#2901](https://github.com/akka/akka-http/pull/2901)
+
+#### akka-http-testkit
+
+* Allow `RouteTest` to run integration tests via `~!>` operator[#](https://github.com/akka/akka-http/pull/)
```
Comment (jrudolph) [inline docs/src/main/paradox/release-notes/10.2.x.md:50]: PR ID missing

```suggestion
* Allow `RouteTest` to run integration tests via `~!>` operator [#3014](https://github.com/akka/akka-http/pull/3014)
```

File: docs/src/main/paradox/release-notes/10.2.x.md [hunk 5]
```diff
@@ -1,14 +1,83 @@
...
+* Update to Scalatest 3.1.0[#2851](https://github.com/akka/akka-http/pull/2851)
```
Comment (jrudolph) [inline docs/src/main/paradox/release-notes/10.2.x.md:51]:
```suggestion
* Update to Scalatest 3.1.0 [#2851](https://github.com/akka/akka-http/pull/2851)
```

File: docs/src/main/paradox/release-notes/10.2.x.md [hunk 6]
```diff
@@ -1,14 +1,83 @@
...
+* Show `RequestBuilding` in client examples [#2968](https://github.com/akka/akka-http/pull/2968)
+* Update documentation and examples to Akka 2.6 [#2996](https://github.com/akka/akka-http/pull/2996)
+* Link to major/minor Akka docs [#3048](https://github.com/akka/akka-http/pull/3048)
+* Make Case class extraction example work with 2.13 [#3092](https://github.com/akka/akka-http/pull/3092)
+* Update extractClientIP.md wrt `remote-address-attribute` [#3083](https://github.com/akka/akka-http/pull/3083)
+* Make `HttpServerWithActorsSample` more 2.6-style [#3077](https://github.com/akka/akka-http/pull/3077)
+
+#### akka-http2-support
+
+* Potential fix for idle timeouts in http2[#2776](https://github.com/akka/akka-http/pull/2776)
```
Comment (jrudolph) [inline docs/src/main/paradox/release-notes/10.2.x.md:68]:
```suggestion
* Potential fix for idle timeouts in http2 [#2776](https://github.com/akka/akka-http/pull/2776)
```

File: docs/src/main/paradox/release-notes/10.2.x.md [hunk 7]
```diff
@@ -1,14 +1,83 @@
...
+* Fix HeaderCompression updating table size without giving notice to peer[#](https://github.com/akka/akka-http/pull/)
```
Comment (jrudolph) [inline docs/src/main/paradox/release-notes/10.2.x.md:69]:
```suggestion
* Fix HeaderCompression updating table size without giving notice to peer [#2891](https://github.com/akka/akka-http/pull/2891)
```

File: docs/src/main/paradox/release-notes/10.2.x.md [hunk 8]
```diff
@@ -1,14 +1,83 @@
...
+
+#### akka-http-caching
+
+* Harden ExpiringLfuCacheSpec [#2960](https://github.com/akka/akka-http/pull/2960)
+
+#### build
+
+* Fix project-info links to API docs [#2857](https://github.com/akka/akka-http/pull/2857)
+* Drop Scala 2.11 [#2589](https://github.com/akka/akka-http/pull/2589)
+* Better diagnostics when validatePullRequest fails weirdly[#](https://github.com/akka/akka-http/pull/)
```
Comment (jrudolph) [inline docs/src/main/paradox/release-notes/10.2.x.md:82]:
```suggestion
* Better diagnostics when validatePullRequest fails weirdly [#2904](https://github.com/akka/akka-http/pull/2904)
```

File: docs/src/main/paradox/release-notes/10.2.x.md (hunk 3 above)
Comment (raboof) [reply docs/src/main/paradox/release-notes/10.2.x.md:12]: I guess we could duplicate the list of highlights from the announcement?

Comment (jrudolph) [reply docs/src/main/paradox/release-notes/10.2.x.md:12]: Yes, exactly, only saw these afterwards.

File: docs/src/main/paradox/migration-guide/migration-guide-10.2.x.md [hunk 9]
```diff
@@ -4,7 +4,21 @@
...
 
-## Akka HTTP 10.1.11 - > 10.2.0
+Under these guidelines, minor version updates are supposed to be binary compatible and drop-in replacements
+for former versions under the condition that user code only uses public, stable, non-deprecated API. Especially
+libraries should make sure not to depend on deprecated API to be compatible with both 10.1.x and 10.2.x.
+
+If you find an unexpected incompatibility please let us know, so we can check whether the incompatibility is accidental so we might still be able to fix it.
```
Comment (ennru) [inline docs/src/main/paradox/migration-guide/migration-guide-10.2.x.md:11]: 👍

**Pull request merged**